### PR TITLE
blog post about QuestDB and crypto

### PR DESCRIPTION
**Pull request by argshook**
This PR adds a new blog post page
`Query a live crypto data stream in millisecs`

### CODE REVIEW

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+## Last prices of BTC and ETH
+
+This query returns the latest price of both BTC-USD and ETH-USD. We use the syntax `LATEST BY` native to QuestDB&#39;s SQL.
+
+Click [here](https://demo.questdb.io/?query=select%20%2a%20from%20trades%0Alatest%20by%20symbol%20%0AWHERE%20symbol%20in%20%28%27BTC-USD%27%2C%20%27ETH-USD%27%29%3B) to fill the query directly on our SQL console with live data.
+
+![](/img/blog/2022-04-07/last-prices-of-btc-and-eth-screenshot.png)
+
+Below is the chart for Bitcoin and Ethereum prices with a time sample of 10 seconds
```
Comment (argshook) [inline blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md:48]: @tris0laris wdyt?

```suggestion
Below is the chart for Bitcoin and Ethereum prices with a time sample of 10 seconds. It is live Grafana chart you can interact with.
```

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+<iframe src="https://dashboard.questdb.io/d-solo/624FG0snk/public-dashboard-1?orgId=1&panelId=10" width="100%" height="300" frameborder="0"></iframe>
```
Comment (argshook) [inline blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md:50]: @tris0laris wdyt, should grafana charts have auto refresh?

```suggestion
<iframe src="https://dashboard.questdb.io/d-solo/624FG0snk/public-dashboard-1?orgId=1&panelId=10&refresh=10s" width="100%" height="300" frameborder="0"></iframe>
```

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+Let's go through these sample queries one by one.  
+
+## Last prices of BTC and ETH
+
+To find out the latest prices of BTC and ETH in USD. We use the syntax
+`LATEST BY`, which is native to QuestDB's SQL Engine.
+
+Click
+[here](https://demo.questdb.io/?query=select%20%2a%20from%20trades%0Alatest%20by%20symbol%20%0AWHERE%20symbol%20in%20%28%27BTC-USD%27%2C%20%27ETH-USD%27%29%3B)
```
Comment (puzpuzpuz) [inline blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md:60]: This query should be:
```sql
SELECT *
FROM trades
WHERE symbol in ('BTC-USD', 'ETH-USD')
LATEST ON timestamp PARTITION BY symbol;
```

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+and time-series SQL extensions in QuestDB. When clicking on a query, it's
+automatically added to the SQL editor. Then, click the &quot;Run&quot; button or
+press <kbd>F9</kbd> to execute the query.
+
+Let's go through these sample queries one by one.  
+
+## Last prices of BTC and ETH
+
+To find out the latest prices of BTC and ETH in USD. We use the syntax
+`LATEST BY`, which is native to QuestDB's SQL Engine.
```
Comment (puzpuzpuz) [inline blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md:57]:
```suggestion
[`LATEST ON`](https://questdb.io/docs/reference/sql/latest-on) syntax, which is native to QuestDB's SQL Engine.
```

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+QuestDB web console. These pre-written queries leverage the standard SQL syntax
+and time-series SQL extensions in QuestDB. When clicking on a query, it's
+automatically added to the SQL editor. Then, click the &quot;Run&quot; button or
+press <kbd>F9</kbd> to execute the query.
+
+Let's go through these sample queries one by one.  
+
+## Last prices of BTC and ETH
+
+To find out the latest prices of BTC and ETH in USD. We use the syntax
```
Comment (puzpuzpuz) [inline blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md:56]:
```suggestion
To find out the latest prices of BTC and ETH in USD. We use the
```

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md (hunk 3 above)
Comment (pswu11) [reply blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md:60]: @puzpuzpuz Thanks! @argshook guess we should also update the example query in the demo box?

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md (hunk 1 above)
Comment (pswu11) [reply blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md]: @argshook I included info about Grafana in the intro paragraphs. wdyt?

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md (hunk 3 above)
Comment (argshook) [reply blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md]: Thanks! Updated the URL with suggested query.
The `Example Queries` popup should also be updated, that effort is tracked separately.

Marking as resolved, please reopen if there's anything left

Comment (puzpuzpuz) [reply blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md]: Thanks!

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+```
+
+Click [here](https://demo.questdb.io/?query=SELECT%20%2a%0AFROM%20trades%0AWHERE%20symbol%20in%20%28%27BTC-USD%27%2C%20%27ETH-USD%27%29%0ALATEST%20ON%20timestamp%20PARTITION%20BY%20symbol%3B)
+to fill the pre-written query directly in the SQL editor, then run the query.
+
+Below is the chart for Bitcoin and Ethereum prices with a time sample of 10
+seconds:
+
+<iframe
+  src="https://dashboard.questdb.io/d-solo/624FG0snk/public-dashboard-1?orgId=1&panelId=10"
```
Comment (argshook) [inline blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md:75]:
```suggestion
  src="https://dashboard.questdb.io/d-solo/624FG0snk/public-dashboard-1?orgId=1&panelId=10&refresh=5s"
```

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+Click
+[here](https://demo.questdb.io/?query=SELECT%20timestamp%2C%20first%28price%29%20as%20open%2C%20last%28price%29%20as%20close%2C%20min%28price%29%2C%20max%28price%29%2C%20sum%28amount%29%20as%20volume%0AFROM%20trades%0AWHERE%20symbol%20%3D%20%27BTC-USD%27%20and%20timestamp%20%3E%20dateadd%28%27d%27%2C%20-1%2C%20now%28%29%29%0ASAMPLE%20BY%2015m%20ALIGN%20TO%20CALENDAR%20WITH%20OFFSET%20%2700%3A00%27%3B)
+to fill the pre-written query directly in the SQL editor, then run the query.
+
+This real-time chart on Grafana plots the candle chart with a time sample of 10
+seconds. We also show the volume traded on a secondary axis.
+
+<iframe
+  src="https://dashboard.questdb.io/d-solo/624FG0snk/public-dashboard-1?orgId=1&panelId=8"
```
Comment (argshook) [inline blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md:108]:
```suggestion
  src="https://dashboard.questdb.io/d-solo/624FG0snk/public-dashboard-1?orgId=1&panelId=8"
```

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+Click
+[here](https://demo.questdb.io/?query=SELECT%20timestamp%2C%20sum%28price%20%2a%20amount%29%20%2F%20sum%28amount%29as%20vwap_price%2C%20sum%28amount%29%20as%20volume%20%0AFROM%20trades%0AWHERE%20symbol%20%3D%20%27BTC-USD%27%20and%20timestamp%20%3E%20dateadd%28%27d%27%2C%20-1%2C%20now%28%29%29%0ASAMPLE%20BY%2015m%20ALIGN%20TO%20CALENDAR%20WITH%20OFFSET%20%2700%3A00%27%3B)
+to fill the pre-written query directly in the SQL editor, then run the query.
+
+The following chart displays the distribution of trades based on their size and
+paints a more granular picture of volume traded.
+
+<iframe
+  src="https://dashboard.questdb.io/d-solo/624FG0snk/public-dashboard-1?orgId=1&panelId=12"
```
Comment (argshook) [inline blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md:139]:
```suggestion
  src="https://dashboard.questdb.io/d-solo/624FG0snk/public-dashboard-1?orgId=1&panelId=12&refresh=5s"
```

File: blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+)
+SELECT btc.timestamp btc_time, btc.price btc_price, eth.price eth_price, round(btc.price/eth.price, 3) btc_to_eth_ratio
+FROM btc
+ASOF JOIN eth;
+```
+
+The following chart plots three series: BTC-USD, ETH-USD and the implied BTC-ETH cross price.
+
+<iframe
+  src="https://dashboard.questdb.io/d-solo/624FG0snk/public-dashboard-1?orgId=1&panelId=14"
```
Comment (argshook) [inline blog/2022-04-04-query-a-live-crypto-data-stream-in-millisecs.md:174]:
```suggestion
  src="https://dashboard.questdb.io/d-solo/624FG0snk/public-dashboard-1?orgId=1&panelId=14&refresh=5s"
```

**Review by argshook — COMMENTED** [review]
add `&refresh=5s` to each iframe `src`

**Pull request merged**